### ISSUE TITLE
Files can incorrectly be nested inside other files

### ISSUE THREAD

**Issue by NotWearingPants** [streamich/memfs#1049]
It's possible to read/write files nested inside other files:

```js
const { fs } = require('memfs');
fs.writeFileSync('/foo', 'hello');
fs.writeFileSync('/foo/bar', 'world');
console.log(fs.readFileSync('/foo', 'utf8')); // logs 'hello'
console.log(fs.readFileSync('/foo/bar', 'utf8')); // logs 'world'
console.log(fs.statSync('/foo').isFile()); // logs 'true'
console.log(fs.statSync('/foo').isDirectory()); // logs 'false'
console.log(fs.readdirSync('/foo')); // throws ENOTDIR
```

`/foo` is a regular file, and `bar` is also a regular file that's nested in `/foo`.

Tested with v4.11.1

**Comment by G-Rath** [comment]
Yup that's a bug alright

**Comment by Kenny4297** [comment]
Hi @NotWearingPants, I think I may be able to help with this issue. I would love to be assigned to it if possible!

**Comment by G-Rath** [comment]
@Kenny4297 go for it!

**Comment by BadIdeaException** [comment]
@G-Rath This is fixed in 4.12.0. This issue can be closed now.

**Comment by G-Rath** [comment]
@BadIdeaException do you know if we've got a test to guard against future regression?

**Comment by BadIdeaException** [comment]
Yes, we do: [openSync.test.ts](https://github.com/streamich/memfs/blob/5460fce836c323ee96a4caeaedcbe1fb10cd4318/src/__tests__/volume/openSync.test.ts#L10) tests this exact scenario.

**Comment by NotWearingPants** [comment]
(For future reference: this issue was fixed by #1059)
